### PR TITLE
refactor: extract forms module

### PR DESCRIPTION
**Pull request by JoomFX**
fixes https://github.com/telerik/kendo-themes/issues/1224

**TODO:**
- [x] Import the Forms module so it is actually compiled
- [x] Move invalid styling to every module (autocomplete, maskedtextbox, dateinput, dropdown...). Change the `$forms-invalid-color` variable to something more appropriate.
- [x] Move "Style helpers" in the Input module
- [x] Refactor Forms module in Material theme
- [x] Move `.k-form` (and all other `.k-form-...`) classes to Form module
- [x] Merge the Edit-Form module into the Forms module

### CODE REVIEW

File: packages/default/scss/forms/_theme.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+@include exports( "forms/theme" ) {
+    .k-widget.k-state-invalid,
```
[the remark this answers is not in the document]
Comment (JoomFX) [reply packages/default/scss/forms/_theme.scss]: Оn the surface of it, it seems that we do not need it 🙂

[the remark this answers is not in the document]
Comment (PreslavKozovski) [reply packages/default/scss/forms/_theme.scss]: I assume we need it - https://github.com/telerik/kendo-themes/commit/497921cee32c4d65014a3ebab8319181db72307c

I believe we need a stronger selector for these styles

[the remark this answers is not in the document]
Comment (JoomFX) [reply packages/default/scss/forms/_theme.scss]: I tested it without the `k-widget` selector and the visual tests did not complain.
But we might need more specificity for cases/scenarios not covered in the visual tests.

[the remark this answers is not in the document]
Comment (PreslavKozovski) [reply packages/default/scss/forms/_theme.scss]: I tested the scenario with an invalid jQuery form, and it looks like we can remove the `.k-widget` selector.

@joneff, do you know a scenario that could regress if we remove this selector?

[the remark this answers is not in the document]
Comment (joneff) [reply packages/default/scss/forms/_theme.scss]: I don't know of any such cases. If we need the class in the future, we can add it on per-component basis

[the remark this answers is not in the document]
Comment (JoomFX) [reply packages/default/scss/forms/_theme.scss]: It turned out that we need `k-widget` for NumericTextbox (more specificity is needed for the invalid styling in the Material theme).

**Pull request merged**